### DISCUSSION TITLE
Remix and KeystoneJS

### DISCUSSION THREAD

**Discussion by phifa** [keystonejs/keystone#8261] [Feature requests]
This would be my dream stack. Have them both work together. 

- Using the powerful KeystoneJS admin interface with the easy schema building in code
- Using the Node API in the remix loaders to query and mutate the data

Anybody ever done this? You would still have to spin up two apps locally, remix and KeystoneJS for the admin UI. I read this recent blog post, this goes in the right direction: https://keystonejs.com/blog/nextjs-keystone

**Comment by acoreyj** [comment]
Yup, I'm not sure if this will be my final solution as I don't want to pay 4 different services if it grows but I have keystone running on railway pointing at a postgresql database from neon and I access it using Prisma client with data proxy from remix running on cloudflare pages

**Comment by TheRealFlyingCoder** [comment]
I'm running a multi-app monorepo setup with kestone and 3 Remix apps

One "parent" remix app handles authentication, and keystone + the 2 other apps each run in sub-domains sharing the parent session cookie

It was a task to get set up and although Keystone doesn't support yarn PNP which makes the monorepo annoying it does work nicely

**Comment by jezzzm** [comment]
I'm running two production apps with Keystone+Remix. Works a charm. not a monorepo though so a bit of extra work to get the schemas synced.

Keystone +postgres -> fly.io
Remix -> vercel (though could also be fly) queries keystone using graphql-request. could go direct to db but taking advantage of keystone's excellent access control so i dont need to reinvent in remix

**Reply by TheRealFlyingCoder** [reply]
Outside of a monorepo, if you have keystone running locally you can use graphql-codegen to generate schema off the localhost url.

You won't have access to the prisma schema though if you need that, other than just copying it across manually haha

Keystone is one of the times where you need a monorepo to keep the dev experience nice and smooth
